src/icalendar/prop.py

```diff
@@ -323,6 +323,11 @@ def to_ical(self):
         else:
             raise ValueError(f'Unknown date type: {type(dt)}')
 
+    def __eq__(self, other):
```

Comment:
If you override `__eq__`, please also override `__hash__` or something like this or the dictionary mapping will fail.

Reply:
[The 'standard' way of implementing hash](https://docs.python.org/3/reference/datamodel.html#object.__hash__) requires fields compared in eq to be hashable, vDDDTypes.param is of type Parameter and that's a subclass of CaselessDict, neither are hashable. Should I make CaslelssDict/Parameter hashable (for example as per [this](https://stackoverflow.com/questions/1151658/python-hashable-dicts#answer-1151705)) and then implement vDDDTypes's hash in the standard way or would you recommend some other solution?

Reply:
Oh, I actually do not know what would be good to do in this case. I would not like to drag it out because of uncertainty. It solves an issue already. That counts. How would you go forward?

Reply:
Personally, after reading tbe stackoverflow answer, I think making them hashable is not worth making everything hashable as they are not supposed to be used as keys anyway and also easily converted to keys if necessary.
It is good to me now and thanks for having the discussion!